Terminal.Gui/ConsoleDrivers/NetDriver.cs

```diff
@@ -177,7 +177,15 @@ void GetConsoleKey ()
 			ConsoleKeyInfo newConsoleKeyInfo = default;
 
 			while (true) {
-				ConsoleKeyInfo consoleKeyInfo = Console.ReadKey (true);
+				ConsoleKeyInfo consoleKeyInfo;
+
+				try {
+					consoleKeyInfo = Console.ReadKey (true);
+				} catch (InvalidOperationException ex) {
```

Comment:
Just eating the exception?

This seems like a bad idea.

Any redirection of Console.In at any point can cause this on a call to `Console.ReadKey()`, and you don't want to eat that exception then, because it isn't recoverable by the library and represents a breaking change.

Reply:
There is no redirection anywhere. Please can you confirm that? Unless is the console API that redirect itself. Ignoring the exception the application continue running normally.

Reply:
In user code.

We are a library, not an application.

And eating exceptions is bad behavior, regardless. We don't know the potential cause of the exception. If we did, we would avoid it or react to it, not eat it.

Reply:
This is easy to react to when we understand why this happens and how to deal with it. This exception is complained about by many users, just do a Google search to confirm it. In my opinion, I think it is because `StdIn` is being used by the terminal API used at the time `Console.ReadKey(bool)` is running. In case of the exception, I have no idea what to do other than ignore it and continue. At least, it has worked well that way. If you know what causes this and can interact with the error, it would be an advantage to explain how to resolve it. I know it's not good practice to ignore exceptions but there are many APIs that generate them and the user already knows that this will happen but they don't need to do anything other than return or ignore and let it flow.

Reply:
I chose to remove all `InvalidOperationException` and use `Task.Delay.Wait` with `CancellationTokenSource` to wait until there is some `ConsoleKeyInfo` available or the `CancellationTokenSource` is cancelled. The cause of the `InvalidOperationException` error is that `Console.ReadKey` cannot be running if `Console.KeyAvailable` is `false`.